### ISSUE TITLE
[Bug]: The number of entities after backup recovery is inconsistent with the number of entities in the source collection, and no delete or upsert operations have been performed on the collection

### ISSUE THREAD

**Issue by zhuwenxing** [zilliztech/milvus-backup#475]
### Current Behavior

The number of entities after backup recovery is inconsistent with the number of entities in the source collection, and no delete or upsert operations have been performed on the collection

```

[2024-12-11T02:49:20.220Z] FAILED testcases/test_restore_backup.py::TestRestoreBackup::test_milvus_restore_back_with_new_feature_support[True-True-True] - AssertionError: srs_num: 3000, dist_num: 2497

[2024-12-11T02:49:20.220Z] FAILED testcases/test_restore_backup.py::TestRestoreBackup::test_milvus_restore_back_with_new_feature_support[True-False-True] - AssertionError: srs_num: 3000, dist_num: 2503

[2024-12-11T02:49:20.220Z] FAILED testcases/test_restore_backup.py::TestRestoreBackup::test_milvus_restore_back_with_new_feature_support[True-False-False] - AssertionError: srs_num: 3000, dist_num: 1503

[2024-12-11T02:49:20.220Z] FAILED testcases/test_restore_backup.py::TestRestoreBackup::test_milvus_restore_back_with_new_feature_support[False-True-True] - AssertionError: srs_num: 3000, dist_num: 1497

[2024-12-11T02:49:20.220Z] FAILED testcases/test_restore_backup.py::TestRestoreBackup::test_milvus_restore_back_with_new_feature_support[False-True-False] - AssertionError: srs_num: 3000, dist_num: 0

[2024-12-11T02:49:20.220Z] FAILED testcases/test_restore_backup.py::TestRestoreBackup::test_milvus_restore_back_with_new_feature_support[False-False-True] - AssertionError: srs_num: 3000, dist_num: 1994
```

### Anything else?

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/milvus_backup_api_test/detail/milvus_backup_api_test/93/pipeline

cluster: 4am
namespace: chaos-testing

milvus
```
[2024-12-11T02:49:28.954Z] + kubectl get pods -o wide

[2024-12-11T02:49:28.955Z] + grep backup-test-93

[2024-12-11T02:49:28.955Z] backup-test-93-etcd-0                                      1/1     Running            0                 10m     10.104.26.43    4am-node32   <none>           <none>

[2024-12-11T02:49:28.955Z] backup-test-93-etcd-1                                      1/1     Running            0                 10m     10.104.25.177   4am-node30   <none>           <none>

[2024-12-11T02:49:28.955Z] backup-test-93-etcd-2                                      1/1     Running            0                 10m     10.104.21.198   4am-node24   <none>           <none>

[2024-12-11T02:49:28.955Z] backup-test-93-kafka-0                                     1/1     Running            2 (9m38s ago)     10m     10.104.25.178   4am-node30   <none>           <none>

[2024-12-11T02:49:28.955Z] backup-test-93-kafka-1                                     1/1     Running            2 (9m40s ago)     10m     10.104.26.45    4am-node32   <none>           <none>

[2024-12-11T02:49:28.955Z] backup-test-93-kafka-2                                     1/1     Running            2 (9m34s ago)     10m     10.104.21.199   4am-node24   <none>           <none>

[2024-12-11T02:49:28.955Z] backup-test-93-milvus-standalone-7cb8d7d45d-8ncbr          1/1     Running            2 (9m53s ago)     10m     10.104.23.122   4am-node27   <none>           <none>

[2024-12-11T02:49:28.955Z] backup-test-93-minio-859d9bfbdf-drnmk                      1/1     Running            0                 10m     10.104.25.175   4am-node30   <none>           <none>

[2024-12-11T02:49:28.955Z] backup-test-93-zookeeper-0                                 1/1     Running            0                 10m     10.104.25.176   4am-node30   <none>           <none>

[2024-12-11T02:49:28.955Z] backup-test-93-zookeeper-1                                 1/1     Running            0                 10m     10.104.26.44    4am-node32   <none>           <none>

[2024-12-11T02:49:28.955Z] backup-test-93-zookeeper-2                                 1/1     Running            0                 10m     10.104.21.197   4am-node24   <none>           <none>
```

backup
```
milvus-backup-93-68894bf558-75xq4   1/1     Running   0          2m1s
```

**Comment by bigsheeper** [comment]
same reason as https://github.com/zilliztech/milvus-backup/issues/478
![image](https://github.com/user-attachments/assets/0f11976d-3f2b-4217-9a8a-03b224751e70)
![image](https://github.com/user-attachments/assets/28c55cce-01bd-482a-8b80-ea7a45b924ca)

**Comment by huanghaoyuanhhy** [comment]
fixed by #482
